File: clean-code/Python/6/6_3.py
from datetime import datetime

def send_email(to, subject, body):
    email = {
        'to': to,
        'subject': subject,
        'body': body,
        'sent_at': datetime.now(),
        'status': 'sent'
    }
    log_email(email)
    return email

welcome_email = send_email("newuser@example.com", "Welcome!", "Thank you for joining.")
welcome_email['is_welcome_email'] = True
log_email(welcome_email)

password_reset_email = send_email("user@example.com", "Reset Password", "Click here to reset your password.")
password_reset_email['is_password_reset'] = True
log_email(password_reset_email)

File: clean-code/C#/6/6_3.cs
using System;

void SendEmail(string to, string subject, string body)
{
    Email email = new Email
    {
        To = to,
        Subject = subject,
        Body = body,
        SentAt = DateTime.Now,
        Status = "sent"
    };
    LogEmail(email);
}

Email welcomeEmail = new Email();
SendEmail("newuser@example.com", "Welcome!", "Thank you for joining.");
welcomeEmail.IsWelcomeEmail = true;
LogEmail(welcomeEmail);

Email passwordResetEmail = new Email();
SendEmail("user@example.com", "Reset Password", "Click here to reset your password.");
passwordResetEmail.IsPasswordReset = true;
LogEmail(passwordResetEmail);
